exercises/robot-simulator/tests/robot-simulator.rs

```diff
@@ -0,0 +1,147 @@
+extern crate robot_simulator;
+
+use robot_simulator::*;
+
+#[test]
+fn robots_are_created_with_position_and_direction() {
+    let robot = Robot::new(0, 0, &Direction::North);
+    assert_eq!((0, 0), robot.position());
+    assert_eq!(&Direction::North, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn positions_can_be_negative() {
+    let robot = Robot::new(-1, -1, &Direction::South);
+    assert_eq!((-1, -1), robot.position());
+    assert_eq!(&Direction::South, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn turning_right_does_not_change_position() {
+    let robot = Robot::new(0, 0, &Direction::North).turn_right();
+    assert_eq!((0, 0), robot.position());
+}
+
+#[test]
+#[ignore]
+fn turning_right_from_north_points_the_robot_east() {
+    let robot = Robot::new(0, 0, &Direction::North).turn_right();
+    assert_eq!(&Direction::East, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn turning_right_from_east_points_the_robot_south() {
+    let robot = Robot::new(0, 0, &Direction::East).turn_right();
+    assert_eq!(&Direction::South, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn turning_right_from_south_points_the_robot_west() {
+    let robot = Robot::new(0, 0, &Direction::South).turn_right();
+    assert_eq!(&Direction::West, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn turning_right_from_west_points_the_robot_north() {
+    let robot = Robot::new(0, 0, &Direction::West).turn_right();
+    assert_eq!(&Direction::North, robot.direction());
+}
+
+#[test]
+#[ignore]
+fn turning_left_does_not_change_position() {
+    let robot = Robot::new(0, 0, &Direction::North).turn_left();
+    assert_eq!((0, 0), robot.position());
+}
+
+#[test]
+#[ignore]
+fn turning_left_from_north_points_the_robot_west() {
+    let robot = Robot::new(0, 0, &Direction::North).turn_left();
```

Comment:
In my testing (which may have missed something) if we want to allow mutable implementations, then the tests you suggest don't work.

For example, if I do this test:

```rust
fn turning_right_does_not_change_position() {
    let robot = Robot::new(0, 0, &Direction::North);
    let robot = robot.turn_right();
    assert_eq!((0, 0), robot.position());
}
```

and this implementation

```rust
pub fn turn_right(&mut self) -> &mut Self {
        self.position = Position::new(1, 1);
        self
}
```

Then I get the compiler error

```
tests/robot-simulator.rs:8:17: 8:22 error: cannot borrow immutable local variable `robot` as mutable
tests/robot-simulator.rs:8     let robot = robot.turn_right();
                                           ^~~~~
tests/robot-simulator.rs:7:9: 7:14 help: to make the local variable mutable, use `mut` as shown:
tests/robot-simulator.rs:      let mut robot = Robot::new(0, 0, &Direction::North);
```

And the compiler is right, if I change the test to

```rust
let mut robot = Robot::new(0, 0, &Direction::North);
```

Then we're cool. 

The current test structure does already allow for the first two function signatures. But I don't know of a way to also allow mutability without mutability being declared in the tests.